trucks.js

```diff
@@ -503,9 +541,92 @@ var foodTrucks = [
   }
 ];
 
-// this module should support the following methods:
-// getTrucks() - return all trucks
-// getTruck(name) - return the truck object matching 'name'
-// getFoodTypes() - return unique list of all associated food types (underscore has a function to help)
-// filterByDay(day) - return trucks with 'day' in schedule (use your filterByDay function from Module 3 homework)
-// filterByFoodType(foodType) - return trucks with associated 'foodType'
\ No newline at end of file
+function getTrucks() {
+  return foodTrucks;
+}
+
+function getTruck( day ) {
+
+  var truck;
+
+  function getTruck( obj ) {
+
+    if ( obj.name && obj.name.indexOf( day ) > -1 ) {
+      return true;
+    } else {
+      return false;
+    }
+  }
+
+  truck = foodTrucks.filter( getTruck );
+
+  return truck;
+}
+
+function getFoodTypes() {
+
+  var i = 0;
+  var l = foodTrucks.length;
+  var foodList = [];
+  var added = {};
+
+  for ( i; i < l; i++ ) {
+    var j = 0;
+    var k = foodTrucks[i].type.length;
+    var type = foodTrucks[i].type;
+
+    for ( j; j < k; j++ ) {
+      var foodItem = type[j].toLowerCase();
+      if ( !added.hasOwnProperty(foodItem)) {
+        foodList.push( type[j] );
+        added[foodItem] = true;
+      }
+    }
+  }
+
+  foodList.sort();
+
+  return foodList;
+}
+
+function filterByDay( day ) {
+
+  var trucks;
+
+  function getTruck( obj ) {
+
+    if ( obj.schedule && obj.schedule.indexOf( day ) > -1 ) {
+      return true;
+    } else {
+      return false;
+    }
+  }
+
+  trucks = foodTrucks.filter( getTruck );
+
+  return trucks;
+}
+
+function filterByFoodType( food ) {
+
+  var trucks;
+
+  function getTruck( obj ) {
+
+    if ( obj.type && obj.type.indexOf( food ) > -1 ) {
+      return true;
+    } else {
+      return false;
+    }
+  }
+
+  trucks = foodTrucks.filter( getTruck );
+
+  return trucks;
+}
+
+module.exports.getTrucks = getTrucks;
```

Comment:
If you want to export all these distinctly, I recommend using the `exports` helper instead as it's more idiomatic than using `module.exports` multiple times. If you want to use `module.exports`, then I would just use it once to export an object with all the methods.